examples/digitalocean/rke.tf

```diff
@@ -1,34 +1,79 @@
-module "nodes" {
+variable "do_token" {
+  default = ""
+}
+
+module "masters" {
+  source = "./do"
+  do_token = var.do_token
+  # region       = "nyc1"
+  # droplet_size = "t2.micro"
+}
+
+module "workers" {
   source = "./do"
+  do_token = var.do_token
   # region       = "nyc1"
   # droplet_size = "t2.micro"
 }
 
 resource "rke_cluster" "cluster" {
   nodes {
```

Comment:
When using PublicIP for the host name of the node, it did not work well. It worked better to specify the name of the droplet in `hostname_override` instead. Were you get well with this setup?

Reply:
it works well with private ip. I can test with `hostname_override`  if needed

Reply:
hmm, I updated and tried again.
But I got some errors.

The following was output to the cloud-controller-manager's log:

```
E0523 07:07:56.269112       1 node_controller.go:417] NodeAddress: Error fetching by providerID: providerID cannot be empty string Error fetching by NodeName: instance not found
```

FYI: I tried with following tf file then I got well.

```
# rke.tf
  nodes {
    hostname_override = module.nodes.names[3]
    internal_address  = module.nodes.internal_addresses[3]
    address           = module.nodes.addresses[3]
    user              = module.nodes.ssh_username
    ssh_key           = module.nodes.private_key
    role              = ["worker"]
  }

# do/outputs.tf
output "names" {
  value = digitalocean_droplet.rke-node.*.name
}
```

Reply:
hmm, I wonder why it works, droplet hostnames in DO not resolvable as I know

Reply:
that's why I set
`kubelet-preferred-address-types = "InternalIP,ExternalIP,Hostname" ` for kube-api server

Reply:
I think it is bug of digitalocean-cloud-controller-manager.

When I specified `kubelet-preferred-address-types = "InternalIP,ExternalIP,Hostname"` to kube-apiserver, cloud-controller-manager could receive droplet's private IP, but ccm seems that couldn't handle it.

I posted the issue to digitalocean-cloud-controller-manager.
I will keep this PR `pending` state until that issue is resolved.

Reply:
`digitalocean-cloud-controller-manager` has been fixed in [v0.1.15](https://github.com/digitalocean/digitalocean-cloud-controller-manager/releases/tag/v0.1.15).

Could you try modifying it to use this version?

Reply:
like this?

Reply:
Thanks for quick response:+1:

I'm going try it.

Reply:
It worked well!!!

Reply:
Cool